api/users/serializers.py

```diff
@@ -52,6 +52,17 @@ def absolute_url(self, obj):
     def update(self, instance, validated_data):
         assert isinstance(instance, User), 'instance must be a User'
         for attr, value in validated_data.items():
-            setattr(instance, attr, value)
+            if 'social' == attr:
+                for key, val in value.items():
+                    instance.social[key] = val
```

Comment:
Probably doesn't need to be fixed here as its a bigger issue, but we should consider running this through a validation map